modules/katello_content_credential.py

```diff
@@ -0,0 +1,167 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_content_credential
+short_description: Create and Manage Katello content credentials
+description:
+  - Create and Manage Katello content credentials
+author: "Baptiste Agasse (@bagasse)"
+requirements:
+  - "nailgun >= 0.31.0"
+  - "python >= 2.6"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+     - Username on Foreman server
+    required: true
+  password:
+    description:
+     - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    default: true
+    type: bool
+  name:
+    description:
+      - Name of the GPG key
+    required: true
+  organization:
+    description:
+      - Organization name that the content credential is in
+    required: true
+  content_type:
+    description:
+    - Type of credential
+    choices:
+    - gpg_key
+    - cert
+    required: true
+  content:
+    description:
+    - Content of the content credential
+    required: true
+  state:
+    description:
+      - State of the content credential.
+    default: present
+    choices:
+      - present
+      - absent
+'''
+
+EXAMPLES = '''
+- name: "Create katello client GPG key"
+  katello_content_credential:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "RPM-GPG-KEY-my-repo"
+    type: gpg_key
+    organization: "Default Organization"
+    content: "{{ lookup('file', 'RPM-GPG-KEY-my-repo') }}"
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+        ContentCredential,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_organization,
+        find_content_credential,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+    has_import_error = False
+except ImportError as e:
+    has_import_error = True
+    import_error_msg = str(e)
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'organization': 'organization',
+    'content_type': 'content_type',
+    'content': 'content',
+}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            name=dict(required=True),
+            organization=dict(required=True),
+            content_type=dict(choices=['gpg_key', 'cert']),
```

Comment:
Docs say `required=True` ?

nailgun defaults to gpg_key, so technically it isn't required but I think I'd prefer if it was

Reply:
Sorry I updated the doc part but forgot this one. For me  nailgun defaults to gpg_key to stay compatible with older versions. Here this is the first version :) . As you I prefer if it is required.